src/Provider/DecoratingPhpFilesSettingsProvider.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Helis\SettingsManagerBundle\Provider;
+
+use Helis\SettingsManagerBundle\Model\DomainModel;
+use Helis\SettingsManagerBundle\Model\SettingModel;
+use Symfony\Component\Cache\Adapter\PhpFilesAdapter;
+use Symfony\Component\Cache\CacheItem;
+use Symfony\Component\Lock\Factory;
+use Symfony\Component\Lock\Store\FlockStore;
+
+class DecoratingPhpFilesSettingsProvider implements ModificationAwareSettingsProviderInterface
+{
+    public const MODIFICATION_TIME_KEY = 'settings_modification_time';
+
+    /** @var SettingsProviderInterface */
+    private $decoratingProvider;
+
+    /** @var PhpFilesAdapter */
+    private $cache;
+
+    /** @var Factory */
+    private $lockFactory;
+
+    /** @var int */
+    private $checkValidityInterval;
+
+    public function __construct(
+        ModificationAwareSettingsProviderInterface $decoratingProvider,
```

Comment:
Will it work if i setup two providers with this decorator at the same time? I think they will be sharing this cache :thinking: shouldn't they have separate cache directories or cache namespace?

Reply:
Resolved by accepting cache AdapterInterface.